File: apprefurbished/src/main/java/com/androidcollider/easyfin/main/MainModule.java
package com.androidcollider.easyfin.main;

import com.androidcollider.easyfin.common.managers.accounts_info.AccountsInfoManager;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class MainModule {

    @Provides
    MainMVP.Model provideMainMVPModel(AccountsInfoManager accountsInfoManager) {
        return new MainModel(accountsInfoManager);
    }

    @Provides
    MainMVP.Presenter provideMainMVPPresenter(MainMVP.Model model) {
        return new MainPresenter(model);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/main/MainModule.kt
package com.androidcollider.easyfin.main

import com.androidcollider.easyfin.common.managers.accounts.accounts_info.AccountsInfoManager
import dagger.Module
import dagger.Provides

/**
 * @author Ihor Bilous
 */
@Module
class MainModule {
    @Provides
    fun provideMainMVPModel(accountsInfoManager: AccountsInfoManager?): MainMVP.Model {
        return MainModel(accountsInfoManager)
    }

    @Provides
    fun provideMainMVPPresenter(model: MainMVP.Model?): MainMVP.Presenter {
        return MainPresenter(model)
    }
}